Library/src/Systems/ECommerce/ECommerce.cpp

```diff
@@ -189,6 +189,175 @@ void ProductInfoDtoToProductInfo(const chs_aggregation::ShopifyProductDto& Dto,
+{
+	ProductInfoCollection = common::Array<csp::systems::ProductInfo>(ProductInfoCollection.Size());
+	int TotalVariantIndex = 0; // Count the total number of variants included in the product Dto's
+	
+	for (int DtoCount = 0; DtoCount < DtoArray.size(); DtoCount++) // Loop all Dto's (products)
+	{
+		const chs_aggregation::ShopifyProductDto& Dto = DtoArray[DtoCount];
+
+		if (Dto.HasVariants()) // if there are no variants, we don't process the dto (shouldn't happen)
+		{
+			for (int VariantCount = 0; VariantCount < Dto.GetVariants().size(); VariantCount++) // Loop each variant in the product dto and store the info into the indexed output array
+			{
+				auto VariantProductInformation = Dto.GetVariants()[VariantCount];
+
+				ProductInfoCollection[TotalVariantIndex].Id = Dto.GetId();
+				ProductInfoCollection[TotalVariantIndex].Title = Dto.GetTitle();
+
+				if (Dto.HasCreatedAt())
+				{
+					ProductInfoCollection[TotalVariantIndex].CreatedAt = Dto.GetCreatedAt();
+				}
+
+				if (Dto.HasDescription())
+				{
+					ProductInfoCollection[TotalVariantIndex].Description = Dto.GetDescription();
+				}
+
+				// The output should only contain productinfo with single variants in, so we hardcode the index and size of the array here
+				ProductInfoCollection[TotalVariantIndex].Variants = common::Array<ProductVariantInfo>(1); 
+				ProductInfoCollection[TotalVariantIndex].Variants[0].Id = VariantProductInformation->GetId();
+				ProductInfoCollection[TotalVariantIndex].Variants[0].Title = VariantProductInformation->GetTitle();
+
+				if (VariantProductInformation->HasImage())
+				{
+					auto VariantProductImage = VariantProductInformation->GetImage();
+
+					if (VariantProductImage->HasMediaContentType())
+					{
+						ProductInfoCollection[TotalVariantIndex].Variants[0].Media.MediaContentType = VariantProductImage->GetMediaContentType();
+					}
+
+					if (VariantProductImage->HasAlt())
+					{
+						ProductInfoCollection[TotalVariantIndex].Variants[0].Media.Alt = VariantProductImage->GetAlt();
+					}
+
+					if (VariantProductImage->HasUrl())
+					{
+						ProductInfoCollection[TotalVariantIndex].Variants[0].Media.Url = VariantProductImage->GetUrl();
+					}
+
+					if (VariantProductImage->HasWidth())
+					{
+						ProductInfoCollection[TotalVariantIndex].Variants[0].Media.Width = VariantProductImage->GetWidth();
+					}
+
+					if (VariantProductImage->HasHeight())
+					{
+						ProductInfoCollection[TotalVariantIndex].Variants[0].Media.Height = VariantProductImage->GetHeight();
+					}
+				}
+				else
+				{
+					CSP_LOG_MSG(LogLevel::Log, "ShopifyProductDto missing Image");
+				}
+
+				if (VariantProductInformation->HasSelectedOptions())
+				{
+					auto VariantOptionInformation = VariantProductInformation->GetSelectedOptions();
+
+					ProductInfoCollection[TotalVariantIndex].Variants[0].Options = common::Array<VariantOptionInfo>(VariantOptionInformation.size());
+
+					for (int n = 0; n < VariantOptionInformation.size(); ++n)
```

Comment:
You have inconsistent placement of the increment expression. Here and below it is prefixed, above for VariantCount it is postfixed. Also is there a reason you are calling the incremented variable 'n' here and 'j' below? There is no 'i' that I can see.